### PR TITLE
[fix](txn insert) sub txn map is not thread safe

### PR DESCRIPTION
**Pull request by mymeiyi**
## Proposed changes

The sub txn map in DatabaseTransactionMgr is not thread safe

### CODE REVIEW

**Comment by mymeiyi** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java [hunk 1]
```diff
@@ -2830,13 +2830,24 @@ private PublishResult checkQuorumReplicas(TransactionState transactionState, lon
     }
 
     protected void addSubTransaction(long transactionId, long subTransactionId) {
-        subTxnIdToTxnId.put(subTransactionId, transactionId);
+        writeLock();
+        try {
+            subTxnIdToTxnId.put(subTransactionId, transactionId);
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java:2835]: `subTxnIdToTxnId` is a concurrent map, is it necessary to lock it?

Comment (mymeiyi) [reply fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java:2835]: it's not concurrent map `private final Map<Long, Long> subTxnIdToTxnId = Maps.newHashMap()`

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]

**Pull request merged**